core/src/lib.rs

```diff
@@ -100,6 +112,9 @@ cfg_if! {
     } else if #[cfg(all(feature = "i-am-libstd", not(target_arch = "wasm32")))] {
         compile_error!("Not allowed to fall back to generic spin lock based thread parker");
     } else {
+        // REVIEW: this implementation isn't really appropriate for most use
+        // cases other than "at least it compiles", and for libstd we'll want a
+        // compile error if this is ever used.
```

Comment:
I believe the above `compile_error!` does precisely wat you want, except for allowing wasm to use the generic impl.

Reply:
That would work, yes, but wasm should continue to not use the generic impl since it should panic if a request to wait is ever received because it won't otherwise wake up.

Reply:
So wasm should do what? Currently the wasm `ThreadParker` requires the `atomics` feature that is not available by default. Should we add a non-atomics wasm `ThreadParker`? How would we implement that? My understanding was that the `atomics` feature was needed in order to do any parking. Or do you mean adding a wasm thread parker that just panics? In other words, making the library compile, but not usable?

Reply:
Well this should compile on CI with both atomics and without atomics. The nonatomic version should just panic if it ever requests to block, and the atomics version should do what it does today. It's difficult to run this so it's fine to just check it compiles. There's example CI configuration in wasm-bindgen of how to set up an atomics target